### PR TITLE
feat: Added modal component

### PR DESCRIPTION
**Pull request by tanishq20**
Added modal variants

- Alert Modal
- Simple Modal
- Confirmation Modal

Live preview link: https://deploy-preview-18--compact-ui.netlify.app/docs/components/modal
preview image

### CODE REVIEW

**Review by srejitk — COMMENTED** [review]
Add inline comments for better readability, and group component styles together. Otherwise looks good

File: stylesheets/components/modal.css [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  font-size: 1.6rem;
+  font-weight: 500;
+  padding: 2rem 0rem;
+  color: var(--black-500);
+}
+
+.modal-alert-action {
+  display: flex;
+  justify-content: flex-end;
+  flex-wrap: wrap;
```
Comment (shubhamkumar648) [inline stylesheets/components/modal.css:33]: you can also use flex flow  :row wrap to reduce code it is horthand property

File: stylesheets/components/modal.css [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  font-weight: 600;
+  color: var(--black-500);
+}
+
+/* Confirmation Modal */
+
+.modal-confirmation-container {
+  background-color: var(--white);
+  width: min(100%, 30rem);
+  height: 300px;
```
Comment (shubhamkumar648) [inline stylesheets/components/modal.css:86]: use rem  units for better responsive

**Review by shubhamkumar648 — COMMENTED** [review]
reamining code looks good to me

File: stylesheets/components/modal.css (hunk 2 above)
Comment (tanishq20) [reply stylesheets/components/modal.css:86]: I want that height to be fixed in a small device also. That's why used px.

File: stylesheets/components/modal.css (hunk 1 above)
Comment (tanishq20) [reply stylesheets/components/modal.css:33]: `flex-flow` works with `flex-direction` and `flex-wrap` that's why I skipped that.

Comment (shubhamkumar648) [reply stylesheets/components/modal.css:33]: okay

File: stylesheets/components/modal.css (hunk 2 above)
Comment (shubhamkumar648) [reply stylesheets/components/modal.css:86]: okay fine

**Review by shubhamkumar648 — APPROVED** [review]
code looks good to merge

**Pull request merged**